devtools-html/debugger.html
# [CallStack] - Refactor and test existing frame grouping behavior

Associated Issue: #5150 

This precursor PR includes some refactoring and adds tests to pave the way for the implementation, which will be the next PR. 

### Summary of Changes
* Moved frame grouping logic so that it runs in the selector, instead of the component render function.
* Added some tests to cover the existing grouping behavior.

### Test Plan

1. Debug the [React example](http://firefox-dev.tools/debugger-examples/examples/my-app/build/).
2. Set a breakpoint on App.js line 14. *React and Webpack frames still grouped as expected in the call stack area.*
4. Open the call stack context menu and click "Disable framework grouping." *Frames are ungrouped as expected.*
5. Re-enable framework grouping. *Frames are grouped as expected*

### Screenshots
This is what the call stack looks like when breaking in an async function transformed by Babel.

Before an `await`:
![image](https://user-images.githubusercontent.com/7321311/36346454-e2a10288-140c-11e8-8e49-af1ba0c909d0.png)

After an `await`:
![image 1](https://user-images.githubusercontent.com/7321311/36346456-e9abb85c-140c-11e8-9af3-3a3e8727747c.png)
